clippy_lints/src/trait_bounds.rs

```diff
@@ -70,6 +70,35 @@ declare_clippy_lint! {
     "Check if the same trait bounds are specified twice during a function declaration"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for multiple instances of the same where clause or trait bound
+    ///
+    /// ### Why is this bad?
+    /// Adds to code noise
+    ///
+    /// ### Example
+    /// ```rust
+    /// fn foo<T: Default + Default>(bar: T) {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo<T: Default>(bar: T) {}
+    /// ```
+    ///
+    /// ```rust
+    /// fn foo<T>(bar: T) where T: Default + Default {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// fn foo<T>(bar: T) where T: Default {}
+    /// ```
+    #[clippy::version = "1.62.0"]
+    pub REPEATED_WHERE_CLAUSE_OR_TRAIT_BOUND,
+    nursery,
+    "Traits are repeated within trait bounds or where clause"
```

Comment:
This should be lower case, just like lint messages. I also noticed that this is also wrong for the other lints in this file. If you want you can also change this for those.
```suggestion
    "traits are repeated within trait bounds or where clause"
```

Reply:
Fixed in https://github.com/rust-lang/rust-clippy/pull/8703/commits/6e2ec70ce3a8ed0f15e33bafdcee400e22bf725c.